include/deal.II/non_matching/mapping_info.h

```diff
@@ -192,7 +192,7 @@ namespace NonMatching
     {
       const auto   jac_0 = inverse_jacobians[0];
       const double zero_tolerance_double =
-        1. / diameter * std::numeric_limits<double>::epsilon() * 1024.;
```

Comment:
What is the value in `MatrixFree`?

Reply:
https://github.com/dealii/dealii/blob/7869e58a5faf7223d44c761c797b7853e2114532/include/deal.II/matrix_free/mapping_info.templates.h#L429-L430

Basically the value before but we look at the Jacobian here not the inverse.

Reply:
> Basically the value before but we look at the Jacobian here not the inverse.

Stupid question: why is used in one case and the other in the other case?

Reply:
The `MatrixFreeFunctions::MappingInfo` computes the data from raw quantities, thus it has the Jacobian at hand. The `NonMatching::MappingInfo` object goes through `Mapping::fill_fe_values` and lets it compute information. I think for `dim != spacedim`, the latter is more general and duplicating the logic would be more work, so the non-matching case simply lets the function compute whatever we eventually consume. One probably should see if there is a way to unify those two cases, but for now I think this is ok.

A consequence is that the inverse Jacobian is less accurate than the Jacobian because the inversion amplifies the roundoff errors.

Reply:
Not a stupid question at all. We currently only request `update_inverse_jacobians` to be set when `update_gradients` is required https://github.com/dealii/dealii/blob/223772e237f56e7527d22c346f12f35a7c27889e/include/deal.II/non_matching/mapping_info.h#L781-L783

This call https://github.com/dealii/dealii/blob/223772e237f56e7527d22c346f12f35a7c27889e/include/deal.II/non_matching/mapping_info.h#L788-L790
then usually also sets `update_jacobians`. We could unconditionally require `update_jacobians`, similar to `internal::MatrixFreeFunctions::MappingInfo` and introduce an additional `UpdateFlags` variable for what should be stored in the end.